source/common/upstream/load_balancer_impl.cc

```diff
@@ -24,9 +24,13 @@ static const std::string RuntimeZoneEnabled = "upstream.zone_routing.enabled";
 static const std::string RuntimeMinClusterSize = "upstream.zone_routing.min_cluster_size";
 static const std::string RuntimePanicThreshold = "upstream.healthy_panic_threshold";
 
-static constexpr uint32_t UnhealthyStatus = 1u << static_cast<size_t>(Host::Health::Unhealthy);
-static constexpr uint32_t DegradedStatus = 1u << static_cast<size_t>(Host::Health::Degraded);
-static constexpr uint32_t HealthyStatus = 1u << static_cast<size_t>(Host::Health::Healthy);
+constexpr uint32_t singleHealthStatusToUint(Host::Health status) {
+  return 1u << static_cast<size_t>(status);
```

Comment:
@htuch 🤔 We don't have complex bit operations here, so there doesn't seem to be any particular benefit to using `std::bitset`.

Reply:
It's not about complexity of bit operations, more about doing this in a more native C++ way; ideally use the right ADTs rather than bit twiddling, which should be less error prone.

Reply:
get it.